### PR TITLE
DG-35 Add logout function

### PR DESCRIPTION
**Pull request by Mickss**

### CODE REVIEW

File: src/auth/SignInComponent.js [hunk 1]
```diff
@@ -1,91 +1,109 @@
...
-            </Link>
-          </Grid>
-          <Grid item>
-            <Link href="#" variant="body2">
-              {"Don't have an account? Sign Up"}
-            </Link>
-          </Grid>
-        </Grid>
-      </Box>
+      {!isLoggedIn ? (
```
Comment (tknowak) [inline src/auth/SignInComponent.js:47]: This part should be reverted. We should not check `isLoggedIn` here. It was moved to Header component.

**Review by tknowak — CHANGES_REQUESTED** [review]

File: src/auth/AuthContext.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { createContext, useState } from "react";
+
+const AuthContext = createContext();
+
+function AuthProvider({ children }) {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [token, setToken] = useState(null);
```
[the remark this answers is not in the document]
Comment (Mickss) [reply src/auth/AuthContext.js:7]: Fixed

File: src/auth/SignInComponent.js (hunk 1 above)
Comment (Mickss) [reply src/auth/SignInComponent.js:47]: Fixed

File: src/Header.jsx [hunk 3]
```diff
@@ -1,21 +1,48 @@
...
           <Button color="inherit">Contact</Button>
         </Box>
 
-        <Button color="inherit" component={ Link } to="/sign-in">Sign in</Button>
+        {!isLoggedIn && (
+          <Button color="inherit" component={Link} to="/sign-in">
+            Sign in
+          </Button>
+        )}
+        {isLoggedIn && (
```
[the remark this answers is not in the document]
Comment (Mickss) [reply src/Header.jsx:41]: Fixed

**Review by tknowak — APPROVED** [review]

**Pull request merged**